happi/audit.py

```diff
@@ -0,0 +1,424 @@
+"""
+This module defines the ``happi audit`` command line utility
+
+If the --file option is not provided, then it will take the path from
+Happi configuration file using the Client.find_config method,
+otherwise it will use the path specified after --file
+"""
+from abc import ABCMeta, abstractmethod
+from configparser import ConfigParser
+import logging
+import os
+import sys
+import subprocess
+import re
+from happi.client import Client
+from happi.loader import import_class, fill_template
+from happi.containers import registry
+from enum import Enum
+
+logger = logging.getLogger(__name__)
+
+
+class ReportCode(Enum):
+    """
+    Class to define report codes
+    """
+    SUCCESS = 0
+    INVALID = 1
+    MISSING = 2
+    EXTRAS = 3
+    NO_CODE = 9
+
+
+class Command(object, metaclass=ABCMeta):
+    """
+    Bluprint for the command class
+    This will be useful if all the commands are inheriting from it
+    """
+
+    def __init__(self, name, summary):
+        self.name = name
```

Comment:
Functionality should be removed in the ABC

Reply:
Ok, took the constructor out.